datasets/consumer-finance-complaints/consumerComplaints.py

```diff
@@ -0,0 +1,240 @@
+    "ID prepaid card",
+    "Mobile wallet",
+    "Government benefit payment card",
+    "Pawn loan",
+    "Other special purpose card",
+    "Check cashing",
+    "Credit repair",
+    "Traveler’s/Cashier’s checks",
+    "Transit card",
+    "Student prepaid card",
+    "Electronic Benefit Transfer / EBT card",
+    "",
+]
+
+_PRODUCT_CATEGORIES = [
+    "Credit reporting, credit repair services, or other personal consumer reports",
+    "Debt collection",
+    "Mortgage",
+    "Credit card or prepaid card",
+    "Checking or savings account",
+    "Credit reporting",
+    "Student loan",
+    "Money transfer, virtual currency, or money service",
+    "Credit card",
+    "Vehicle loan or lease",
+    "Bank account or service",
+    "Payday loan, title loan, or personal loan",
+    "Consumer Loan",
+    "Payday loan",
+    "Money transfers",
+    "Prepaid card",
+    "Other financial service",
+    "Virtual currency",
+]
+
+
+class consumerComplaints(datasets.GeneratorBasedBuilder):
+    """Consumer Complaints dataset."""
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "Date Received": datasets.Value("timestamp[s]"),
+                "Product": datasets.features.ClassLabel(names=_PRODUCT_CATEGORIES),
+                "Sub Product": datasets.features.ClassLabel(names=_SUB_PRODUCT_CATEGORIES),
+                "Issue": datasets.Value("string"),
+                "Sub Issue": datasets.Value("string"),
+                "Complaint Text": datasets.Value("string"),
+                "Company Public Response": datasets.Value("string"),
+                "Company": datasets.Value("string"),
+                "State": datasets.Value("string"),
+                "Zip Code": datasets.Value("string"),
+                "Tags": datasets.features.ClassLabel(names=_TAGS),
+                "Consumer Consent Provided": datasets.Value("string"),
+                "Submitted via": datasets.Value("string"),
+                "Date Sent To Company": datasets.Value("string"),
+                "Company Response To Consumer": datasets.Value("string"),
+                "Timely Response": datasets.Value("string"),
+                "Consumer Disputed": datasets.Value("string"),
+                "Complaint ID": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+            task_templates=[
+                TextClassification(text_column="Complaint Text", label_column="Product"),
+                TextClassification(text_column="Complaint Text", label_column="Sub Product"),
+                TextClassification(text_column="Complaint Text", label_column="Tags"),
+            ],
+        )
+
+    def _split_generators(self, dl_manager):
+        path = dl_manager.download_and_extract(_url)
+        path = f"{path}/complaints.csv"
```

Comment:
Please use `os.path.join` or you might have issues on windows (the separator is a backslash, not a slash) or in streaming mode (we use [URL chaining](https://filesystem-spec.readthedocs.io/en/latest/features.html#url-chaining))
```suggestion
        path = os.path.join(path, "complaints.csv")
```